### DISCUSSION TITLE
Old flux1 + flux2 on cluster.

### DISCUSSION THREAD

**Discussion by slarktits** [fluxcd/flux2#2855] [Q&A] [answered]
Hello.
During infrastructure updating and migration I want to use both version of flux in same cluster.

I have old flux1 inside namespace "flux" with connected repos app1, app2
and New flux2 inside namespace "flux-system" with connected repos app3, app4

Is it possible for both versions to exist in this form?

**Comment by kingdonb** [comment, marked as answer]
Yes, you can run Flux v1 and Flux v2 on the same cluster.

If you are using Helm Operator, I'd recommend upgrading to Helm Controller everywhere first. There is extremely limited support available for both Flux v1 and Helm Operator, I have tested them together with Flux v2 in limited ways only, our goal is to get people to migrate and not leave anyone running Flux v1, so it can be sunset and EOL. There is still limited support available for Flux v1 issues, you can see exactly how long we have been beating this drum here at the [Migration Timetable](https://fluxcd.io/docs/migration/timetable/) document.

The current phase of support began June 30, 2021, in it we ask all users or projects that depend on Flux to migrate to Flux v2. If you have reasons why you cannot migrate every workload, we believe that feature parity was achieved already in the previous phase, so if you have still reasons why Flux v2 is not able to handle every workload, they should be filed as bugs.

Issues related to misbehavior when you are running Flux v1 and Flux v2 in a combined setting, off the top of my head I can think of one annoying thing that's easy to fix, but requires a breaking change so you'll have to do it on your own:

`kubectl get HelmReleases` has two resolved CRDs if you have both Helm Operator and Flux v2 (with Helm Controller) installed together on the same cluster, so when you run `kubectl get helmreleases` without thinking or fully qualifying you have to guess which one you're looking at. Worse, `kubectl get hr` is likely to return the Helm Operator's `helmreleases.helm.fluxcd.io`.

I really don't recommend using Helm Operator at all at this point, as the support document explicitly says `Flux 1 Helm Operator code freeze – no further updates except CVEs` – but there are breaking changes in some upstreams which blocked our upgrades, that means some dependencies are likely to contain CVEs or other issues that are known and cannot currently be mitigated in a non-breaking way.

Should be nothing stopping you from using Helm Controller together with Flux v1, for example, although I can't say with certainty about any specifics about how I've tested it. If it's not clear, I'm really interested to know what's keeping you on Flux v1, so we can determine if it's a common issue with no workaround, then we can prioritize to address it with changes in Flux v2 if needed.

Flux v1 support may continue for some time in its current form, we are working proactively on security issues, but please be clear from our side and security there is no security in Flux v1 by design, and you should not use it if you want your cluster to be secure today.

Some of Flux v2's breaking changes from Flux v1 are specific with regard to the shelling out and exec of binaries that can be said to have lead to vulnerabilities before including cluster-admin. Flux v1 made bad decisions in this area which Flux v2 does not do this shelling out, Flux v1 did and can be exploited for sure, if you are enabling Kustomize and Flux v1, which I know you don't have to enable, but it is dangerous to allow these so we recommend Flux v2 with Server-Side Apply instead, it is known to be more secure.

**Reply by slarktits** [reply]
Thank you.
